lumicks/pylake/kymo.py

```diff
@@ -1,43 +1,75 @@
 import warnings
 from copy import copy
 from dataclasses import dataclass
+from lumicks.pylake.detail.image import first_pixel_sample_indices
 
 import numpy as np
 
+from .detail.utilities import method_cache
 from . import colormaps
 from .adjustments import no_adjustment
 from .detail.confocal import (
     ConfocalImage,
     ScanAxis,
     ScanMetaData,
 )
-from .detail.image import histogram_rows, round_down, seek_timestamp_next_line
+from .detail.image import histogram_rows, round_down, seek_timestamp_next_line, InfowaveCode
 from .detail.plotting import get_axes, show_image
 from .detail.timeindex import to_timestamp
 
 
 def _default_line_time_factory(self: "Kymo"):
-    """Line time in seconds"""
-    if self.timestamps.shape[1] > 1:
-        ns_to_sec = 1e-9
+    """Line time in seconds
+
+    The line time is defined as the time between frames (including the dead-time between frames).
+
+    Raises
+    ------
+    RuntimeError
+        If there is only a single scan line in the info wave (this makes it impossible to
+        determine the actual line time).
+    """
+    ns_to_sec = 1e-9
+    single_line_error = (
+        "This kymograph consists of only a single line. It is not possible to determine the "
+        "kymograph line time for a kymograph consisting only of a single line."
+    )
+
+    if self._has_default_factories():
+        infowave = self.infowave  # Make sure we pull this out only once, the slice is not free
+        start, stop = first_pixel_sample_indices(infowave.data)
+        pixel_samples = stop - start + 1
+        end_of_line = start + self.pixels_per_line * pixel_samples
+
+        try:
+            beyond_first_line = infowave.data[end_of_line:] != InfowaveCode.discard
+            dead_time = np.argmax(beyond_first_line)
+
+            # Special case for no dead time. Is it really no deadtime, or a single line?
+            if dead_time == 0 and not np.any(beyond_first_line):
+                raise RuntimeError(single_line_error)
+
+        except ValueError:  # ValueError occurs when beyond_first_line is empty
+            raise RuntimeError(single_line_error)
+
+        return (self.pixels_per_line * pixel_samples + dead_time) * infowave._src.dt * ns_to_sec
```

Comment:
nitpicky, perhaps a tiny bit nicer
```suggestion
    if self._has_default_factories():
        infowave = self.infowave  # Make sure we pull this out only once, the slice is not free
        start, stop = first_pixel_sample_indices(infowave.data)
        pixel_samples = stop - start + 1
        scan_time = self.pixels_per_line * pixel_samples

        try:
            beyond_first_line = infowave.data[start + scan_time :] != InfowaveCode.discard
            dead_time = np.argmax(beyond_first_line)

            # Special case for no dead time. Is it really no deadtime, or a single line?
            if dead_time == 0 and not np.any(beyond_first_line):
                raise RuntimeError(single_line_error)

        except ValueError:  # ValueError occurs when beyond_first_line is empty
            raise RuntimeError(single_line_error)

        return (scan_time + dead_time) * infowave._src.dt * ns_to_sec
```

Reply:
Ah yeah. Amended! 👍